### ISSUE TITLE
Can you help include the pretrained model of MobileSAM in this official project?

### ISSUE THREAD

**Issue by ChaoningZhang** [facebookresearch/segment-anything#454]
We have just released our MobileSAM project to make SAM mobile-friendly. MobileSAM replaces the heavyweight encoder (632M) in the original SAM with a much smaller Tiny-ViT (5M). It performs on par with the original SAM. For inference speed, MobileSAM runs around 10ms per image: 8ms on the image encoder and 2ms on the mask decoder. MobileSAM keeps exactly the same pipeline as the original SAM, and therefore, the MobileSAM (with pre-trained weights) can be integrated into the official SAM project by just changing the image encoder. Thanks for including the pretrained model of MobileSAM in this official project.

**Comment by fakeyee** [comment]
Does the MobileSAM can be used on iOS or Android platform? Is there any Demo, thanks.

**Comment by ryouchinsa** [comment]
Here is the [Segment Anything CPP Wrapper for macOS](https://github.com/ryouchinsa/sam-cpp-macos/tree/master).
This code is originated from [Segment Anything CPP Wrapper](https://github.com/dinglufe/segment-anything-cpp-wrapper) and implemented on macOS app [RectLabel](https://rectlabel.com). We customized the original code so that getMask() uses the previous mask result called as low_res_logits and retain the previous mask array for undo/redo actions. 

We hope this macOS version would help to develop the iOS version.
Please let us know your opinion.

![sam_polygon](https://github.com/ryouchinsa/sam-cpp-macos/assets/1954306/4640e139-c533-4b8c-b27b-e02a401b9bbd)

**Comment by ryouchinsa** [comment]
Improved "Create polygon using SAM" feature so that you can label pixels using the pixels option. You can label [Segment Anything 1 Billion (SA-1B)](https://ai.meta.com/datasets/segment-anything-downloads/) like dataset by yourself.

![sam-pixels](https://github.com/facebookresearch/segment-anything/assets/1954306/d828c18b-6fda-4258-b866-21bc34a0d17f)
